src/KubeLibrary/KubeLibrary.py

```diff
@@ -84,6 +87,55 @@ def __init__(self, kube_config=None, context=None, api_url=None, bearer_token=No
         self.reload_config(kube_config=kube_config, context=context, api_url=api_url, bearer_token=bearer_token,
                            ca_cert=ca_cert, incluster=incluster, cert_validation=cert_validation)
 
+    @staticmethod
+    def get_names_from_resource_list(resource_list):
+        return {item.metadata.name for item in resource_list.items}
+
+    @staticmethod
+    def generate_random_name(size):
+        return "".join(choices(ascii_lowercase + digits, k=size))
+
+    @staticmethod
+    def get_from_k8s_client(attr_name, *args, **kwargs):
+        attr = getattr(client, attr_name, None)
+        assert callable(attr), f"kubernetes.client does not contain {attr_name}!"
+        return attr(*args, **kwargs)
+
+    def get_dynamic_resource(self, api_version, kind):
+        if not self.dynamic_client:
+            self.dynamic_client = dynamic.DynamicClient(client.ApiClient(configuration=client.Configuration().get_default_copy()))
+        return self.dynamic_client.resources.get(api_version=api_version, kind=kind)
+
+    def get(self, api_version, kind, **kwargs):
+        resource = self.get_dynamic_resource(api_version, kind)
+        return resource.get(**kwargs)
+
+    def create(self, api_version, kind, **kwargs):
+        resource = self.get_dynamic_resource(api_version, kind)
+        resource.create(**kwargs)
+
+    def delete(self, api_version, kind, **kwargs):
+        resource = self.get_dynamic_resource(api_version, kind)
+        resource.delete(**kwargs)
+
+    def patch(self, api_version, kind, **kwargs):
+        resource = self.get_dynamic_resource(api_version, kind)
+        resource.patch(**kwargs)
+
+    def replace(self, api_version, kind, **kwargs):
+        resource = self.get_dynamic_resource(api_version, kind)
+        resource.replace(**kwargs)
+
+    def watch(self, api_version, kind, **kwargs):
+        resource = self.get_dynamic_resource(api_version, kind)
+        yield resource.watch(**kwargs)
+
+    @on_predicate(constant, lambda x: x not in ("succeeded", "failed", "unknown"), interval=5, max_time=60)
```

Comment:
I think it would be good to stick to kubernetes library `watch` for it (see https://stackoverflow.com/questions/65938572/kubernetes-python-client-equivalent-of-kubectl-wait-for-command). We could save us from adding another dependency.

Also currently using sth like:
https://github.com/devopsspiral/KubeLibrary/blob/27030048c8facb1f0e7cc6d0359249d7bd414135/testcases/pod/pod_kw.robot#L14

Reply:
Okay I can remove this.

Reply:
So this one is still there

Reply:
Hmm, it's weird. I don't have this in my local version as well as in the PR changed files tab. I will need to check why there is a difference between our versions...